src/library/tabledelegates/keydelegate.cpp

```diff
@@ -0,0 +1,56 @@
+#include "library/tabledelegates/keydelegate.h"
+
+#include <QPainter>
+#include <QStyle>
+#include <QTableView>
+
+#include "moc_keydelegate.cpp"
+#include "track/keyutils.h"
+
+void KeyDelegate::paintItem(
+        QPainter* painter,
+        const QStyleOptionViewItem& option,
+        const QModelIndex& index) const {
+    paintItemBackground(painter, option, index);
+
+    QString keyText = index.data().toString();
+    mixxx::track::io::key::ChromaticKey key = KeyUtils::guessKeyFromText(keyText);
+    const auto keyColor = KeyUtils::keyToColor(key);
+
+    if (keyColor.isValid()) {
+        // Draw the colored rectangle next to the key label
+        painter->fillRect(
+                option.rect.x() + 2,
```

Comment:
I suggest to remove the left padding in order to allow narrow column configurations.
Would look like this
![image](https://github.com/mixxxdj/mixxx/assets/5934199/dfd8973d-1baa-4def-8ad9-647125366a3b)

Reply:
Done. Thank you for this suggestion!